eureka2-core/src/main/java/com/netflix/eureka2/registry/PreservableEurekaRegistry.java

```diff
@@ -194,17 +209,26 @@ public void call() {
 
     @Override
     public Observable<Void> shutdown(Throwable cause) {
+        moveHealthTo(Status.DOWN);
+
         evictionSubscription.unsubscribe();
         evictionQueue.shutdown();
         return eurekaRegistry.shutdown(cause);
     }
 
+    @Override
+    public Status toEurekaStatus(Status healthStatus) {
+        return healthStatus;
+    }
+
     /**
      * FIXME eviction strategies need a rethink
      * >= 0 as when both sizes are equal, we still allow eviction to happen as they may be stale copies
      */
     private boolean allowedToEvict() {
-        return evictionStrategy.allowedToEvict(expectedRegistrySize, eurekaRegistry.size()) >= 0;
+        boolean allowed = evictionStrategy.allowedToEvict(expectedRegistrySize, eurekaRegistry.size()) >= 0;
+        moveHealthTo(allowed ? Status.UP : Status.DOWN);
```

Comment:
Per our discussion, we should not mark servers as UP even when they are in selfPreservation. Otherwise we may redirect all traffic to a (possibly) much smaller set of UP servers and take them down.